spec/unleash/feature_toggle_spec.rb

```diff
@@ -20,16 +20,126 @@
 
   describe 'FeatureToggle with empty strategies' do
     let(:feature_toggle) { Unleash::FeatureToggle.new(
-      name: 'test',
-      enabled: true,
-      strategies: [],
-      variants: nil
+      JSON.parse('{
+          "name": "test",
+          "enabled": true,
+          "strategies": [],
+          "variants": []
+        }')
```

Comment:
While it is possible to `JSON.parse` the string, into the object that the `FeatureToggle` object will use, as a unit test, it would be faster if avoided the deserialization and just sent the hash directly. (as it was done before).

So I don't see the value added in changing this code. For these tests, there is no difference either if `variants` is `nil` or an empty array: `[]`.

Reply:
I test it with `pry` found the feature toggle object not properly initialized as we configured by the `hash`, so I take JSON and seems like it works. Will verify this later.

Reply:
```ruby
let(:feature_toggle) { Unleash::FeatureToggle.new({
        "name" => "test",
        "enabled" => true,
        "strategies" => [],
        "variants" => nil
      }) }

> feature_toggle
=> #<Unleash::FeatureToggle:0x00007f870b264d70 @enabled=true, @name="test", @strategies=[], @variant_definitions=[]>
```
^ guess the hash's key are string type instead of symbol, so this would not work from original code base:

```ruby
let(:feature_toggle) { Unleash::FeatureToggle.new(
        name: "test",
        enabled: true,
        strategies: [],
        variants: nil
      ) }

> feature_toggle
=> #<Unleash::FeatureToggle:0x00007f94b8965ab0 @enabled=false, @name=nil, @strategies=[], @variant_definitions=[]>
```
and this should support the assumption:

https://github.com/Unleash/unleash-client-ruby/blob/master/lib/unleash/feature_toggle.rb#L14..L21

now will replace JSON impl. as hash with string keys instead.

Reply:
set back `variants` value as `nil`

fix on commit 1ca04ac72fd6edcd507860350b28d74d2172bb9f